widgetssdk/src/main/java/com/glia/widgets/secureconversations/SecureConversations.kt

```diff
@@ -42,16 +45,34 @@ class SecureConversationsImpl(
     private val secureConversations: com.glia.androidsdk.secureconversations.SecureConversations
 ) : SecureConversations {
 
-    override fun getUnreadMessageCount(callback: RequestCallback<Int?>) {
-        secureConversations.getUnreadMessageCount(callback)
+    override fun getUnreadMessageCount(onSuccess: OnSuccess<Int>?, onError: OnError?) {
```

Comment:
What about adding the default values?
```suggestion
    override fun getUnreadMessageCount(onSuccess: OnSuccess<Int>? = null, onError: OnError? = null) {
```

Reply:
Agree with `onError`. Here and in other places. However, I don't think it makes sense to specify `null` as a default value for `onSuccess` because what's the point of calling this function then? Am I missing anything? Here is how I've changed it for now dbe3c9aa06b461e1298b1ead52ac26fb6cc2e0a6.

Reply:
I've reworked the implementation. Please have a look now.